ngsderive/commands/junction_annotation.py

```diff
@@ -0,0 +1,301 @@
+                    num_too_few_reads += 1
+                    continue
+                num_novel += 1
+                num_novel_spliced_reads += num_reads
+                if not disable_junction_files:
+                    print(
+                        "\t".join(
+                            [
+                                contig,
+                                str(intron_start),
+                                str(intron_end),
+                                str(num_reads),
+                                annotation,
+                            ]
+                        ),
+                        file=junction_file,
+                    )
+
+            logger.debug(
+                f"{num_too_few_reads} potential junctions didn't have enough read support."
+            )
+            logger.debug(f"{len(events) - num_too_few_reads} junctions annotated.")
+            continue
+
+        collapsed_junctions = defaultdict(int)
+
+        for intron_start, intron_end, num_reads in events:
+            start_novel, ref_start = annotate_event(
+                intron_start, cache.exon_ends[contig], fuzzy_range
+            )
+
+            end_novel, ref_end = annotate_event(
+                intron_end, cache.exon_starts[contig], fuzzy_range
+            )
+
+            if ref_start:
+                start = ref_start
+            else:
+                start = intron_start
+            if ref_end:
+                end = ref_end
+            else:
+                end = intron_end
+
+            if fuzzy_range:
+                collapsed_junctions[(start, end)] += num_reads
+            else:  # only execute if not fuzzy searching
+                if num_reads < min_reads:
+                    num_too_few_reads += 1
+                    continue
+                if start_novel and end_novel:
+                    annotation = "complete_novel"
+                    num_novel += 1
+                    num_novel_spliced_reads += num_reads
+                elif start_novel or end_novel:
+                    annotation = "partial_novel"
+                    num_partial += 1
+                    num_partial_spliced_reads += num_reads
+                else:
+                    annotation = "annotated"
+                    num_known += 1
+                    num_known_spliced_reads += num_reads
+
+                if not disable_junction_files:
+                    print(
+                        "\t".join(
+                            [
+                                contig,
+                                str(start),
+                                str(end),
+                                str(num_reads),
+                                annotation,
+                            ]
+                        ),
+                        file=junction_file,
+                    )
+
+        # if not fuzzy searching, collapsed_junctions is empty and loop is skipped
+        for (intron_start, intron_end), num_reads in sorted(
```

Comment:
Why are you looping over the junctions again?

Reply:
This allows read counts of nearby junctions to be added before the `min_reads` filter. So I need to loop through all returned events, collapse the events that fuzzy-match to the same junction, and then loop again for a final filter before writing out.

Reply:
Ok, I missed that initially looking at this. Makes sense.

Reply:
Just added some comments to make that logic more clear.